simulation/agents.py

```diff
@@ -513,7 +513,9 @@ def update_heating_status(self, model: "DomesticHeatingABM") -> None:
 
         self.reset_previous_heating_decision_log()
 
-        step_interval_years = model.step_interval / datetime.timedelta(days=365)
+        step_interval_years = (
+            model.step_interval.months + (12 * model.step_interval.years) / 12
```

Comment:
Bracketing needs to be updated here, order of operations is problematic

Reply:
Whoops! 🙈